From bfe1e6e631a7f19924e8ce574e04694ebae28bd0 Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Thu, 24 May 2018 16:57:17 +0800
Subject: [PATCH] ASoC: rockchip: i2s_tdm: Cast pointer to uintptr_t to fix
 warning on 32-bit

On 32-bit:
sound/soc/rockchip/rockchip_i2s_tdm.c:155:18: warning:
cast from pointer to integer of different size [-Wpointer-to-int-cast]

Cast the pointer to uintptr_t instead of u64 to fix this.

Change-Id: Ie7dbbb59da68fadfe8ad639d93c513e38ab00b38
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 sound/soc/rockchip/rockchip_i2s_tdm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/rockchip/rockchip_i2s_tdm.c b/sound/soc/rockchip/rockchip_i2s_tdm.c
index 01c820ceacac..f4f3ba01e587 100644
--- a/sound/soc/rockchip/rockchip_i2s_tdm.c
+++ b/sound/soc/rockchip/rockchip_i2s_tdm.c
@@ -145,7 +145,7 @@ static void rockchip_snd_xfer_reset_assert(struct rk_i2s_tdm_dev *i2s_tdm,
 			addr = cru_reset + (rx_bank * 4);
 		}
 
-		if (IS_ALIGNED((u64)addr, 8)) {
+		if (IS_ALIGNED((uintptr_t)addr, 8)) {
 			writeq(val, addr);
 			break;
 		}
@@ -189,7 +189,7 @@ static void rockchip_snd_xfer_reset_deassert(struct rk_i2s_tdm_dev *i2s_tdm,
 			addr = cru_reset + (rx_bank * 4);
 		}
 
-		if (IS_ALIGNED((u64)addr, 8)) {
+		if (IS_ALIGNED((uintptr_t)addr, 8)) {
 			writeq(val, addr);
 			break;
 		}
-- 
2.35.3

